server/src/test/java/org/elasticsearch/common/collect/ImmutableOpenMapTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.collect;
+
+import com.carrotsearch.hppc.cursors.ObjectObjectCursor;
+
+import org.elasticsearch.common.Randomness;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class ImmutableOpenMapTests extends ESTestCase {
+
+    ImmutableOpenMap<String, String> regionCurrencySymbols = ImmutableOpenMap.<String, String>builder()
+        .fPut("Japan", "¥")
+        .fPut("USA", "$")
+        .fPut("EU", "€")
+        .fPut("UK", "£")
+        .fPut("Korea", "₩")
+        .build();
+
+    public void testStreamOperationsAreSupported() {
+        assertThat(regionCurrencySymbols.stream().filter(e -> e.getKey().startsWith("U")).map(Map.Entry::getValue)
+                .collect(Collectors.toSet()), equalTo(Set.of("£", "$")));
+    }
+
+    public void testSortedStream() {
+        assertThat(regionCurrencySymbols.stream().sorted(Map.Entry.comparingByKey()).map(Map.Entry::getValue).collect(Collectors.toList()),
+            equalTo(List.of("€", "¥", "₩", "£", "$")));
+    }
+
+    public void testStreamOperationsOnRandomMap() {
+        ImmutableOpenMap<Long, String> map = Randomness.get().longs(1000)
+            .mapToObj(e -> Tuple.tuple(e, randomAlphaOfLength(8)))
+            .reduce(ImmutableOpenMap.<Long, String>builder(), (builder, t) -> builder.fPut(t.v1(), t.v2()), (a, b) -> b)
```

Comment:
I think this is not safe wrt. the `reduce` contract, since the accumulator mutates its input and the combiner does not combine. Instead you can use the 3 param `collect` which allows mutable reduction. Or collect to a Map and then using `ImmutableOpenMap.Builder.putAll`, can still be done in one line.

Or generate the `sortedMap` first and then use `ImmutableOpenMap.Builder.putAll` to generate the `ImmutableOpenMap`, that avoids the conversion to map below.

Reply:
Good point about the difference between `collect` and `reduce`!